CONTRIBUTING.md

```diff
@@ -1,24 +1,430 @@
+>
+> Additionally, have a look at the [Roadmap] to see Firebase's
+> longer term goals. There are many opportunities to get involved!
+
+### API Review
+Please note that changes or additions to public APIs require an internal API
+review from the Firebase team. Contributions involving such changes will
+require additional time to allow for an internal API review to be scheduled and
+thoroughly conducted. We appreciate your patience while we review your amazing
+contributions!
+
+### Breaking Changes
+Firebase's release schedule is designed to reduce the amount of breaking
+changes that developers have to deal  with. Ideally, breaking changes should
+be avoided when making a contribution.
+
+### Using GitHub pull requests
+All submissions, including submissions by project members, require review. We
+use GitHub pull requests for this purpose. Refer to [GitHub Help] for
+more information on using pull requests. If you're ready to open a pull request,
+check that you have completed all of the steps outlined in
+the [Opening a pull request](#opening-a-pull-request) section.
+
+----
+<!-- Development Guide -->
+
+## Development Guide
+The majority of the remaining portion of this guide is dedicated to detailing
+the preferred workflow for Firebase development.
+
+<!-- ### Touring the codebase -->
+<!-- TODO: Provide a graphic of key areas of the codebase. -->
+
+### Getting started
+To develop Firebase software, **install**:
+- [Xcode] (v12.2 or later) (for Googlers, visit [go/xcode](go/xcode)) to
+  download.
+- <details>
+  <summary>Code styling tools: <b>clang-format</b> & <b>mint</b></summary>
+
+   <!-- The above line is intentionally left blank. -->
+   Firebase use's a style script that requires [clang-format] and [mint].
+
+   To install [clang-format] and [mint] using [Homebrew]:
+    ```console
+    brew install clang-format@12
+    brew install mint
+    ```
+  </details>
+
+**Next**, clone the Firebase repo:
+- Clone via [HTTPS][github-clone-https]
+  ```console
+  git clone https://github.com/firebase/firebase-ios-sdk.git
+  ```
+- Or via [SSH][github-clone-ssh]
+  ```console
+  git clone git@github.com:firebase/firebase-ios-sdk.git
+  ```
+
+Once the necessary tools have been installed and the project has been cloned,
+continue on to the preferred
+[development workflow](#developing).
+
+### Developing
+The workflow for library development is different from application
+development. For Firebase development, we develop using the same tools we
+use to distribute Firebase. Instructions for developing with
+[Swift Package Manager](#swift-package-manager) and [CocoaPods](#cocoapods)
+are as follows:
+
+#### **[Swift Package Manager]**
+[Swift Package Manager] is built into Xcode and makes it simple to develop
+projects with multiple dependencies.
+
+To develop using SwiftPM, simple open the `Package.swift` file in your cloned
+Firebase copy and select a library scheme to build and develop that library.
```

Comment:
"simple open the `Package.swift` file of your cloned Firebase copy in Xcode and"